tt_eager/tensor/tensor.hpp

```diff
@@ -32,10 +32,12 @@ struct Tensor {
         DataType dtype;
         Layout layout;
         std::mutex populated_mutex;
-        std::vector<bool> tensor_populated = {};
+        uint32_t num_shards = 0;
```

Comment:
can we use a better variable name here

Reply:
I will address this in the next commit. Does `num_shards_to_be_populated` seem better? That's effectively what we're tracking here.